### PR TITLE
Adding magic-cta-pipe

### PR DESCRIPTION
**Pull request by aleberti**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jsitarek** [comment]
> @jsitarek @Elisa-Visentin I added you as maintainer of magic-cta-pipe in conda-forge. Could you confirm it is fine with you by commenting this PR?

yes, thanks!

File: recipes/magic-cta-pipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  host:
+    - python >=3.9
+    - setuptools >=64
```
Comment (jsitarek) [inline recipes/magic-cta-pipe/meta.yaml:20]: in MCP there is no lower limit, just setuptools<=71 in environment (and nothing in setup file)

**Comment by Elisa-Visentin** [comment]
Fine for me, thanks

**Comment by aleberti** [comment]
I have to create a new version of the package, since some dependencies are creating problems. I will update the recipe after that.

**Comment by aleberti** [comment]
@conda-forge/help-python the build is failing on win_64, for the following:

```bash
Encountered problems while solving:
  - nothing provides requested protozfits >=2.5,<3.dev0
  - nothing provides protozfits >=2.2,<3.dev0 needed by ctapipe_io_lst-0.22.3-pyhd8ed1ab_0
```

but for example the recipe https://github.com/conda-forge/lstchain-feedstock/blob/main/recipe/meta.yaml has the same requirement and it seems it does not have problems. Also, I see that `protozfits` latest version in conda-forge is 2.5.1 (https://anaconda.org/conda-forge/protozfits). Do you have suggestions?

**Comment by Elisa-Visentin** [comment]
@aleberti as far as I know protozfits does not exist for Windows... Maybe using selectors to exclude it for win_64? But
- probably you could build it, but not use it due to the missing package
- ctapipe_io_lst requires it, too

I just checked ctapipe_io_lst package (and also the one you posted above) and they only have the linux entry in the azure_pipelines... Maybe this is a solution?

**Comment by aleberti** [comment]
> @aleberti as far as I know protozfits does not exist for Windows... Maybe using selectors to exclude it for win_64? But
> 
> * probably you could build it, but not use it due to the missing package
> * ctapipe_io_lst requires it, too
> 
> I just checked ctapipe_io_lst package (and also the one you posted above) and they only have the linux entry in the azure_pipelines... Maybe this is a solution?

Yes I know it is not available on windows, but for some reason I though that for lstchain the conda package was available for windows, but it actually is not.

I just checked the PR when `lstchain` was added (https://github.com/conda-forge/staged-recipes/pull/23429) by Max, and also there it fails on windows. Quoting Max:

> The windows failure is expected, although this is a no-arch package, we have a dependency on a python/c++binding package not available on windows

I cannot access the details of the failure since the build is ~1 year old, but from Max says, it is because of protozfits. Being so, I would simply ask for review now.

**Comment by aleberti** [comment]
@conda-forge/help-python ready for review!

The windows failure is expected, although this is a no-arch package, we have a dependency on a python/c++binding package not available on windows. In case we need to do anything on the recipe, let us know!

**Pull request merged**